config.md

```diff
@@ -147,6 +147,23 @@ flag parsing (clang vs clang-cl), target inference (gcc-arm-noneabi) etc.
 If the option matches a glob mentioned in `--query-driver`, then it'll be
 invoked for extraction of include paths.
 
+### BuiltinHeaders
+{:.v21}
+
+Controls whether Clangd should includes its own built-in headers (like stddef.h), or use the system header found from the query driver.
+
+Valid values are:
+- `Clangd`: Use builtin headers from `clangd`.
+- `QueryDriver`: Use the headers extracted from the compiler via the
```

Comment:
Given the discussion in https://github.com/llvm/llvm-project/pull/129459, could you please also add a sentence like the following:

"Note that if the driver is not clang, this will result in the clang frontend (embedded in clangd) processing the builtin headers of another compiler, which could lead to unexpected results such as false positive diagnostics."

Reply:
Thats written well enough, I think I'll just copy and paste it in :)